pinot-connectors/pinot-spark-3-connector/src/main/scala/org/apache/pinot/connector/spark/v3/datasource/PinotBufferedRecordReader.scala

```diff
@@ -0,0 +1,69 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.connector.spark.v3.datasource
+
+import org.apache.pinot.spi.data.readers.{GenericRow, RecordReader, RecordReaderConfig}
+
+import java.io.File
+import java.util
+
+/**
+ * A buffered record reader that stores the records in memory and allows for iteration over them.
+ * This is useful to satisfy the RecordReader interface in Pinot, and also allow for Spark executor
+ * to write records.
+ *
+ * To improve resilience, we may write records to disk when memory is full,
```

Comment:
Use a TODO to describe such possible improvements in future. Or we can simply delete this line as the feature is not included in this PR.